src/main/java/controllers/ParameterPage.java

```diff
@@ -0,0 +1,58 @@
+package controllers;
+
+import exceptions.NotSupportedEncodingException;
+import request.HTTPRequest;
+import response.HTTPResponse;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.util.HashMap;
+import java.util.Map;
+
+import static response.StatusLine.OK;
+
+public class ParameterPage extends Controller {
+
+    @Override
+    public HTTPResponse get(HTTPRequest request) {
+        return new HTTPResponse(OK, buildBodyWith(decodedParameters(request)));
+    }
+
+    private Map<String, String> decodedParameters(HTTPRequest request) {
+        Map<String, String> decodedParameters = new HashMap<>();
+        if (containsMoreParameters(request)) {
```

Comment:
I find it a bit confusing that this method mutates `decodedParameters` but it's not returned anywhere. So rather than having the methods `decodeAllParameters` and `decodeParameter` as void, maybe return the `decodedParameters` and assign them in this method. So that it's clearer that these methods add something to that HashMap.

Reply:
👍